tasks/db_deploy/create_db.py

```diff
@@ -0,0 +1,144 @@
+from orca_sql import *
+
+
+def create_orca_database():
+    # Check to see if database exists if it does not create the database.
+    # root_connection = shared_db.get_root_connection(config)
+
+    # TODO: Need to fix this so it is not in a transaction
+    # with root_connection.connect() as conn:
+    #    _create_app_database(conn, config["database"], config["root_user"])
+
+    # Check to see if database exists if it does then perform more checks to
+    # determine the next actions to take otherwise create the database, users,
+    # roles and install the schema
+    pass
+
+
+def create_fresh_orca_install(config):
+    """
+    This task will create the ORCA roles, users, schema, and tables needed
+    by the ORCA application as a fresh install.
+
+    Args:
+        config (Dict): Dictionary with database connection information
+    """
+    # Assume the database has been created at this point. Connect to the ORCA
+    # database as a super user and create the roles, users,  schema, and
+    # objects.
+    root_app_connection = get_root_connection(config, config["database"])
+
+    with root_app_connection.connect() as conn:
+        # Create the roles, schema and user
+        create_app_schema_role_users(conn, config)
+
+        # Change to DBO role and set search path
+        set_search_path_and_role(conn)
+
+        # Create the database objects
+        create_metadata_objects(conn)
+        create_recovery_objects(conn)
+
+        # If everything is good, commit.
+        conn.commit()
+
+
+def create_app_schema_role_users(connection, config):
+    """
+    Creates the ORCA application database schema, users and roles.
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
+    """
+    # Create the roles first since they are needed by schema and users
+    logger.debug("Creating the ORCA dbo role ...")
+    connection.execute(dbo_role_sql())
+    logger.info("ORCA dbo role created.")
+
+    logger.debug("Creating the ORCA app role ...")
+    connection.execute(app_role_sql())
+    logger.info("ORCA app role created.")
+
+    # Create the schema next
+    logger.debug("Creating the ORCA schema ...")
+    connection.execute(orca_schema_sql())
+    logger.info("ORCA schema created.")
+
+    # Create the users last
+    logger.debug("Creating the ORCA application user ...")
+    connection.execute(app_user_sql(config["app_user_password"]))
+    logger.info("ORCA application user created.")
+
+
+def set_search_path_and_role(connection):
+    """
+    Sets the role to the dbo role to create/modify ORCA objects and sets the
+    search_path to make the orca schema first. This must be run before any
+    creations or modifications to ORCA objects in the ORCA schema.
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
+    """
+    # Set the user and search_path for the install
+    logger.debug("Changing to the dbo role to create objects ...")
```

Comment:
I'm guessing these steps are quick and minor, and thus do not need a 'completed' log?

Reply:
Correct, I could potentially make it info but, this steps basically changes our "postgres" user so it masquerades as the ORCA dbo role. This way all of the objects are owned by that role. This makes permission management much easier within the schema while also using least privilaged permissions.